### PR TITLE
[Drop Keyspace] Oracle Part 1: Simple refactoring

### PR DESCRIPTION
**Pull request by mdaudali**
## General
**Before this PR**:
We can't drop Oracle "keyspaces" automatically, so there's no easy way to clean up stale data right now.

For reference, the final implementation will look something like: https://github.com/palantir/atlasdb/compare/mdaudali/prototype/oracledeletion?expand=1, of which 1300 lines are tests, the other 400 implementation... (Tests will get cleaned up as I make PRs.)

Specifically for this PR, there's a number of places that require some simple timed sql connection supplier that duplicate code. 
**After this PR**:
Since I need the sql connection supplier for some tests, I've refactored it out. 
As the total feature PR is large, this is it's own PR.

==COMMIT_MSG==
==COMMIT_MSG==

**Priority**: P1

**Concerns / possible downsides (what feedback would you like?)**:
The implementations are identical, except the refactored version doesn't use a protected method. Given nothing in the package relies on it, and it's a package protected method on an anonymous class, this is safe (unless someone is doing scary reflection)
**Is documentation needed?**:
Not at this point
## Compatibility
**Does this PR create any API breaks (e.g. at the Java or HTTP layers) - if so, do we have compatibility?**:
No
**Does this PR change the persisted format of any data - if so, do we have forward and backward compatibility?**:
No
**The code in this PR may be part of a blue-green deploy. Can upgrades from previous versions safely coexist? (Consider restarts of blue or green nodes.)**:
Yes
**Does this PR rely on statements being true about other products at a deployment - if so, do we have correct product dependencies on these products (or other ways of verifying that these statements are true)?**:
No
**Does this PR need a schema migration?**
No
## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**:
None
**What was existing testing like? What have you done to improve it?**:
Unchanged
**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**:
N/A
**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**:
N/A
## Execution
**How would I tell this PR works in production? (Metrics, logs, etc.)**:
DbKVS users don't complain
**Has the safety of all log arguments been decided correctly?**:
N/A
**Will this change significantly affect our spending on metrics or logs?**:
N/A
**How would I tell that this PR does not work in production? (monitors, etc.)**:
DbKvs users complain 
**If this PR does not work as expected, how do I fix that state? Would rollback be straightforward?**:
Rollback
**If the above plan is more complex than “recall and rollback”, please tag the support PoC here (if it is the end of the week, tag both the current and next PoC)**:
N/A
## Scale
**Would this PR be expected to pose a risk at scale? Think of the shopping product at our largest stack.**:
No
**Would this PR be expected to perform a large number of database calls, and/or expensive database calls (e.g., row range scans, concurrent CAS)?**:
No
**Would this PR ever, with time and scale, become the wrong thing to do - and if so, how would we know that we need to do something differently?**:
No
## Development Process
**Where should we start reviewing?**:
It's small, the main thing is to verify that there's actually no change.

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**:
:/

**Please tag any other people who should be aware of this PR**:

### CODE REVIEW

File: atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/SimpleTimedSqlConnectionSupplier.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    private static final class SimpleSql extends SQL {
+        @Override
+        protected SqlConfig getSqlConfig() {
+            return new SqlConfig() {
+                @Override
+                public boolean isSqlCancellationDisabled() {
+                    return false;
+                }
+
+                private Iterable<SqlTimer> getSqlTimers() {
```
Comment (jeremyk-91) [inline atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/SimpleTimedSqlConnectionSupplier.java:62]: nit: Private methods are typically placed below interface methods that actually use them.

**Review by jeremyk-91 — APPROVED** [review]
I think this is fine, though something for awareness:

> Given nothing in the package relies on it, and it's a package protected method on an anonymous class, this is safe (unless someone is doing scary reflection)

Where would you draw the line for reflection being scary? For me this could break a user using read-based reflection, which I wouldn't perceive as scary, as opposed to just unusual. I'd be a fair bit less comfortable with this change if the SQL or SQL config objects are retrievable via the API (though I had a look and they don't seem readily exposed unless internals of things are made visible, which for me does cross the line into questionable/scary reflection).

**Comment by mdaudali** [comment]
My rather blurry line is roughly something like: If you're writing a framework and your users are explicitly opting into using it (e.g annotations, subclassing, i.e, not relying on some magic property existing at runtime), or more vaguely, providing lib users a less bloated and more usable dependency set (say, AtlasDBFactory discovery). 
Or, while still scary but more justifiable: There's an urgent business problem that needs remediating, and you've exhausted all avenues of making upstream API changes, _and_ you've accepted the fact that what you're relying on can break at any point in time.

To counter your point:  `I'd be a fair bit less comfortable with this change if the SQL or SQL config objects are retrievable via the API` - shouldn't we then expose this directly, rather than relying on consumers to do reflection to access something?

In this case: Consumers can't trivially subclass this (it's anonymous), nothing in DbKvsOracleTests rely on this, so it would be consumers of DbKvs. From there, you'd need to extract the SQL buried a few layers into the KVS calls (all of which are non-exposed internal things) and pray. This does seem scary, since we're diving well into non-exposed internal implementation details here!

**Comment by jeremyk-91** [comment]
The answer to that is that sometimes we have power-users who want to do things that we don't want to allow the general user to do (we've done this a lot with internal server framework and timelock, where things like disabling all request logging are knobs we don't generally want to expose but we insist on because of performance reasons). Or there's a disagreement between upstream usage and us 😅 

Agree that SQL is fairly deep in, though note that large internal product uses atlasdb-dbkvs classes directly, and the SqlConnectionSupplier is public API of a DbKVS in that you can just get it (though that's as far as I saw).

I think my idea of scary is more modifying stuff as opposed to just nasty and awkward (if one inspects the object and makes a decision _if_ some property is true, I'd put that in that bucket).

**Pull request merged**